scripts/tests/test_validate_docstrings.py

```diff
@@ -1004,6 +1006,30 @@ def test_item_subsection(self, idx, subsection):
         assert result[idx][3] == subsection
 
 
+class TestDocstringClass(object):
+    @pytest.mark.parametrize('name_and_expected_obj',
+                             [('pandas.isnull', pandas.isnull),
+                              ('pandas.DataFrame', pandas.DataFrame),
+                              ('pandas.Series.sum', pandas.Series.sum)])
+    def test_resolves_class_name(self, name_and_expected_obj):
+        name, expected_obj = name_and_expected_obj
+        d = validate_docstrings.Docstring(name)
+        assert d.obj is expected_obj
+
+    @pytest.mark.parametrize('invalid_name', ['panda', 'panda.DataFrame'])
+    def test_raises_for_invalid_module_name(self, invalid_name):
+        # Note that the module names in this test are misspelled.
```

Comment:
is this ImportError generated by us? or just a generic module name? maybe adding match= here?

Reply:
Yes, this is raised in the validate_docstrings.py script.  That script takes a command-line argument like "pandas.DataFrame.head" or "pandas.isnull" (a string), then dynamically finds the function/method from the string.  The script will then validate docstrings for that function only, which is useful for developers working on a particular docstring.

https://github.com/pandas-dev/pandas/blob/9c0f6a8d703b6bee48918f2c5d16418a7ff736e3/scripts/validate_docstrings.py#L258-L272

I can make the test match the message like so... I don't want to over-specify the message here, so I could also just make sure that the invalid user-supplied is in the exception message.

```python
    @pytest.mark.parametrize('invalid_name', ['panda', 'panda.DataFrame'])
    def test_raises_for_invalid_module_name(self, invalid_name):
        msg = 'No module can be imported.*{}'.format(invalid_name)
        with pytest.raises(ImportError, match=msg):
            validate_docstrings.Docstring(invalid_name)
```